src/cloud/components/atoms/ThreadStatusFilterControl.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useCallback } from 'react'
+import { mdiChevronDown, mdiAlertCircleOutline } from '@mdi/js'
+import { capitalize } from '../../lib/utils/string'
+import {
+  useContextMenu,
+  MenuTypes,
+  MenuItem,
+} from '../../../shared/lib/stores/contextMenu'
+import { Thread } from '../../interfaces/db/comments'
+import Icon, {
+  SuccessIcon,
+  PrimaryIcon,
+  WarningIcon,
+} from '../../../shared/components/atoms/Icon'
+import { RoundButton } from '../../../shared/components/atoms/Button'
+import styled from '../../../shared/lib/styled'
+
+export type StatusFilter = Thread['status']['type'] | 'all'
+
+interface StatusFilterControlProps {
+  value: StatusFilter
+  onChange: (filter: StatusFilter) => void
+  counts: { [K in StatusFilter]: number }
+}
+
+function ThreadStatusFilterControl({
+  value,
+  onChange,
+  counts,
+}: StatusFilterControlProps) {
+  const { popup } = useContextMenu()
+
+  const openActionMenu: React.MouseEventHandler<HTMLButtonElement> = useCallback(
+    (event) => {
+      event.preventDefault()
+      popup(event, buildMenu(onChange, counts))
+    },
+    [popup, onChange, counts]
+  )
+
+  return (
+    <FilterButton variant='icon_secondary' onClick={openActionMenu}>
+      <ThreadFilterIcon status={value} />
+      <span>
+        {capitalize(value)} ({counts[value]})
+      </span>
+      <Icon path={mdiChevronDown} />
+    </FilterButton>
+  )
+}
+
+const FilterButton = styled(RoundButton)`
+  & .button__label {
+    display: flex;
+    align-items: center;
+    min-width: 120px;
+
+    & svg:first-child {
+      margin-right: ${({ theme }) => theme.sizes.spaces.xsm}px;
+    }
+
+    & span {
```

Comment:
I'd like to provide className or to provide a more specific selector like https://github.com/BoostIO/BoostNote.next/pull/962/files#diff-239ca82f07967f8b6edfe509bf554f61460d41c36e06879ab6510b51700a5098R49. This might cause style issue when spans are added or removed in `Icon` or `ThreadFilterIcon` components.